test/util/spawned_process.hpp

```diff
@@ -0,0 +1,71 @@
+/*************************************************************************
+ *
+ * Copyright 2023 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **************************************************************************/
+
+#ifndef REALM_TEST_UTIL_SPAWNED_PROCESS_HPP
+#define REALM_TEST_UTIL_SPAWNED_PROCESS_HPP
+
+#ifdef _WIN32
+#include <windows.h>
+#endif
+
+#include <memory>
+#include <string>
+
+namespace realm {
+namespace test_util {
+
+// This is a wrapper around UNIX fork(), and Windows CreateProcess().
```

Comment:
Is there a a good reason to use fork() on non-Windows? Using posix_spawn() (or even just fork+evec) would make the tests work much more similarly between platforms and would maybe eliminate the need for all the weird stuff to make forking work?

Reply:
I've updated to use `posix_spawn()`